lib/components/result-view/plotly.js

```diff
@@ -0,0 +1,149 @@
+/* @flow */
+/**
+ * Adapted from https://github.com/nteract/nteract/blob/master/packages/transform-plotly/src/index.tsx
+ * Copyright (c) 2016 - present, nteract contributors
+ * All rights reserved.
+ *
+ * This source code is licensed under the license found in the LICENSE file in
+ * the root directory of this source tree.
+ *
+ * @NOTE: This `PlotlyTransform` component could be used exactly same as the original `PlotlyTransform` component of @nteract/transform-plotly,
+ *        except that this file adds the ability to download a plot from an electron context.
+ */
+
+import { cloneDeep } from "lodash";
+import * as React from "react";
+
+interface Props {
+  data: string | Object;
+  mediaType: "application/vnd.plotly.v1+json";
+}
+
+type ObjectType = Object;
+
+interface FigureLayout extends ObjectType {
+  height?: string;
+  autosize?: boolean;
+}
+
+interface Figure extends ObjectType {
+  data: Object;
+  layout: FigureLayout;
+}
+
+declare class PlotlyHTMLElement extends HTMLDivElement {
+  data: Object;
+  layout: Object;
+  newPlot: () => void;
+  redraw: () => void;
+}
+
+const NULL_MIMETYPE = "text/vnd.plotly.v1+html";
+const MIMETYPE = "application/vnd.plotly.v1+json";
+
+/*
+ * As part of the init notebook mode, Plotly sneaks a <script> tag in to load
+ * the plotlyjs lib. We have already loaded this though, so we "handle" the
+ * transform by doing nothing and returning null.
+ */
+const PlotlyNullTransform = () => null;
+PlotlyNullTransform.MIMETYPE = NULL_MIMETYPE;
+PlotlyNullTransform.defaultProps = {
+  mediaType: NULL_MIMETYPE
+};
+
+export class PlotlyTransform extends React.Component<Props> {
+  static MIMETYPE = MIMETYPE;
+
+  static defaultProps = {
+    data: "",
+    mediaType: MIMETYPE
```

Comment:
You could just hardcode the `MIMETYPE` string here, `@nteract/outputs` looks for `mediaType` and only the jupyter extension still uses `MIMETYPE` as far as I know (we don't need it for hydrogen)

The reason I mention it is that [`MIME` is a term that has been replaced with `Media Type`](https://www.iana.org/assignments/media-types/media-types.xhtml).

Reply:
Agreed